### ISSUE TITLE
Helm `tpl` template function

### ISSUE THREAD

**Issue by technosophos** [kubernetes/helm#1978]
There has been some interest in supporting an in-place template feature:

```
{{ tpl "My {{.template}}" . }}
```

And to be able to embed templates into values:

```
{{ tpl .Values.myTemplate . }}
```

And to render templates inside of files:

```
{{ $file := .Files.Get "myfile" }}
{{ tpl $file . }}
```

This is a very tricky case, but can be done in late binding like `include`.

**Comment by syhol** [comment]
This also ties into an issue I created a while back, specifically the comment I left on Jan 16th showing that people are injecting and templating files within the chart. 

https://github.com/kubernetes/helm/issues/1754#issuecomment-272911132

**Comment by lightsaway** [comment]
This would be a great help so you can have separate folder with files , and one wrapper for configmaps in templates folder :
```
kind: ConfigMap
metadata:
  name: my-configmap
data:
{{ tpl (.Files.Glob "files/*").AsConfig | indent 2 }}
```

**Comment by dohnto** [comment]
I believe this one can be closed due to #2350

**Comment by bacongobbler** [comment]
Thanks @dohnto.
